helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -744,14 +767,20 @@ private RollingUpdateOp rollingUpdateAwaitRunning(final ZooKeeperClient client,
 
       // We haven't detected any errors, so assume the agent will write the status soon.
       return opFactory.yield();
-    } else if (!taskStatuses.get(deploymentGroup.getJobId()).getState()
-        .equals(TaskStatus.State.RUNNING)) {
+    } else if (!taskStatuses.get(jobId).getState().equals(TaskStatus.State.RUNNING)) {
       // job isn't running yet
 
       if (isRolloutTimedOut(client, deploymentGroup)) {
         // time exceeding the configured deploy timeout has passed, and this job is still not
         // running
-        return opFactory.error("timed out waiting for job to reach RUNNING", host);
+        String errMsg = String.format("timed out waiting for job %s to reach RUNNING.", jobId);
```

Comment:
Any reason to not just defer this until we have the previous states and construct the error message all in one go?